### PR TITLE
Db-checkbook adding in data loading functions

### PR DESCRIPTION
**Pull request by DeaBardhoshi**
Adding two functions to read in the data from Digital Ocean.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -130,3 +130,6 @@ dmypy.json
 
 .vscode
 *.DS_Store
+
+
+db-checkbook/historical_cpdb_data/
```
Comment (athursland) [inline .gitignore:135]: I think if you upload the input data to the .library file it will be ignored - so maybe we should move the historical cpdb data there? (Although this totally works too)

File: db-checkbook/build_scripts/01_dataloading.py [hunk 2]
```diff
@@ -1,16 +1,39 @@
...
+
+# reading in the checkboob daa
+def read_from_csv(name: str) -> pd.DataFrame:
+    df = pd.read_csv(
+        "https://edm-recipes.nyc3.cdn.digitaloceanspaces.com/datasets/test_nypl_libraries/20210122/test_nypl_libraries.csv", dtype=str, index_col=False
+    )
+    return df
+
+def read_from_gpd(path):
+    df = gpd.read_file('https://edm-recipes.nyc3.cdn.digitaloceanspaces.com/datasets/dcp_cpdb/2018_adopted/dcp_cpdb.sql', geometry = 'geometry')
```
Comment (athursland) [inline db-checkbook/build_scripts/01_dataloading.py:31]: I like naming geopandas df's `gdf` to differentiate but that's also just a personal preference

File: db-checkbook/build_scripts/01_dataloading.py [hunk 3]
```diff
@@ -1,16 +1,39 @@
...
+  #  s3_client = client()
+   # available_buckets = [
+    #    bucket["Name"] for bucket in s3_client.list_buckets()["Buckets"]
+    #]
+   # print(f"This S3 client is a {type(s3_client)}")
+   # print(f"This S3 client has access to buckets: {available_buckets}")
+
+
+# reading in the checkboob daa
+def read_from_csv(name: str) -> pd.DataFrame:
```
Comment (athursland) [inline db-checkbook/build_scripts/01_dataloading.py:24]: maybe we should specify the url on digital ocean as a param as well so this function can be generalized to different csvs (noting that the url is currently a placeholder)

File: db-checkbook/build_scripts/01_dataloading.py [hunk 4]
```diff
@@ -1,16 +1,39 @@
-from dcpy.connectors.s3 import client
+#from dcpy.connectors.s3 import client
```
Comment (athursland) [inline db-checkbook/build_scripts/01_dataloading.py:1]: going off of another comment I think that eventually we'll incorporate the client function from dcpy to grab the data from digital ocean

File: db-checkbook/build_scripts/01_dataloading.py (hunk 3 above)
Comment (fvankrieken) [reply db-checkbook/build_scripts/01_dataloading.py:24]: A couple notes on aligning this with our bash functionality in terms of how its used
- store the "base" url as a literal (i.e. at top of file for now, have `recipes_url="https://...` or something like that
- use name to format string in read_csv
- have `version` as an optional arg with default value `latest`. In this case at the moment, the version is `20210122`

File: db-checkbook/dcp_cpdb.yml
Comment (fvankrieken) [inline db-checkbook/dcp_cpdb.yml:1]: Looks like this is an empty file that can be removed

File: db-checkbook/build_scripts/01_dataloading.py [hunk 5]
```diff
@@ -1,16 +1,40 @@
...
+
+
+
+def read_s3(digital_ocean_filepath, save_file_path):
+
+    #digital_ocean_filepath example "datasets/bpl_libraries/20210208/bpl_libraries.csv"
+
+    s3_client = s3.client()
+    file = s3_client.download_file('edm-recipes', digital_ocean_filepath, save_file_path)
+    return file
```
Comment (fvankrieken) [inline db-checkbook/build_scripts/01_dataloading.py:22]: Looks like this function is more what you're going to use than the read_from_csv? At least in terms of moving a file from s3 to local storage.

Since our recipe file paths are pretty consistent, maybe take dataset name, extension, and version all as arguments similar to my suggestion for `read_from_csv`? And again, version could be optional and default to "latest". And at that point, you could construct the DO_filepath from these arguments.

This could also use a more descriptive name - it's downloading a file specifically from edm-recipes, so name should reflect that.

Lastly, not sure what boto3 `download_file` actually returns ([docs](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/download_file.html) are even unclear) - since this is more a function that has a side-effect than output, I might not return anything (@alexrichey / @damonmcc thoughts?)

File: db-checkbook/build_scripts/02_build.py [hunk 6]
```diff
@@ -1,5 +1,8 @@
 import pandas as pd
 
+def run_categorization(df):
```
Comment (athursland) [inline db-checkbook/build_scripts/02_build.py:3]: best practice might be to open a separate PR for changes made to the build script and keep the scope of this PR as small as possible (limiting to dataloading)

**Comment by fvankrieken** [comment]
Hi @DeaBardhoshi - is this PR ready for review/merging, or are you still working on it?

**Comment by DeaBardhoshi** [comment]
Hi @fvankrieken,  the current functions pulling in data are ready to be reviewed.
